zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,992 @@
+        randomness: F,
+    ) -> Result<(), Error> {
+        layouter.assign_region(
+            || "keccak table",
+            |mut region| {
+                let mut offset = 0;
+
+                // All zero row to allow simulating a disabled lookup.
+                self.keccak_assign_row(&mut region, offset, F::zero(), F::zero(), 0, F::zero())?;
+                offset += 1;
+
+                for aux in &auxs {
+                    let KeccakAux { input, output } = aux;
+                    let input_rlc =
+                        RandomLinearCombination::random_linear_combine(*input, randomness);
+                    let output_rlc = Word::random_linear_combine(*output, randomness);
+                    self.keccak_assign_row(
+                        &mut region,
+                        offset,
+                        F::one(),
+                        input_rlc,
+                        input.len(),
+                        output_rlc,
+                    )?;
+                    offset += 1;
+                }
+                Ok(())
+            },
+        )?;
+        Ok(())
+    }
+
+    pub fn ecc_chip_config(&self) -> EccConfig {
+        EccConfig::new(self.range_config.clone(), self.main_gate_config.clone())
+    }
+
+    pub fn integer_chip_config(&self) -> IntegerConfig {
+        IntegerConfig::new(self.range_config.clone(), self.main_gate_config.clone())
+    }
+}
+
+pub struct AssignedECDSA<F: FieldExt> {
+    pk_x_le: [AssignedValue<F>; 32],
+    pk_y_le: [AssignedValue<F>; 32],
+    msg_hash_le: [AssignedValue<F>; 32],
+}
+
+#[derive(Debug)]
+pub(crate) struct AssignedSignatureVerify<F: FieldExt> {
+    pub(crate) address: AssignedCell<F, F>,
+    pub(crate) msg_hash_rlc: AssignedCell<F, F>,
+}
+
+// Returns assigned constants [256^1, 256^2, .., 256^{n-1}]
+fn assign_pows_256<F: FieldExt>(
+    ctx: &mut RegionCtx<'_, '_, F>,
+    main_gate: &MainGate<F>,
+    n: usize,
+) -> Result<Vec<AssignedValue<F>>, Error> {
+    let mut pows = Vec::new();
+    for i in 1..n {
+        pows.push(main_gate.assign_constant(ctx, F::from(256).pow(&[i as u64, 0, 0, 0]))?);
+    }
+    Ok(pows)
+}
+
+// Return an array of bytes that corresponds to the little endian representation
+// of the integer, adding the constraints to verify the correctness of the
+// conversion (byte range check included).
+fn integer_to_bytes_le<F: FieldExt, W: WrongExt>(
+    ctx: &mut RegionCtx<'_, '_, F>,
+    main_gate: &MainGate<F>,
+    range_chip: &RangeChip<F>,
+    pows_256: &[AssignedValue<F>],
+    int: &AssignedInteger<W, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+) -> Result<[AssignedValue<F>; 32], Error> {
+    let mut int_le = Vec::new();
+    int_le.extend(int.limbs()[0].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[1].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[2].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[3].decompose(5, 8).expect("bad decompose"));
```

Comment:
All the expects that are in non-test code should only fail if there are bugs in the code.  At least this was my aim.
This should be no different than doing `v[4]` which can panic if `v.len() < 5`; but if we know for sure that the length of `v` is `>= 5` we don't do error handling, because we don't expect it to panic.

In this case, the decompose function takes values known at runtime, but the values used here should always return a `Some<T>`.

Reply:
I see! You're completelly correct! My bad here!